### PR TITLE
Complete signup

### PR DESCRIPTION
**Pull request by KitanGarcia**
#### This PR encompasses three areas:
1) The endpoints for deletion of Creators and Benefits    
2) Testing deletion for Creators and Benefits
3) Completion of the Sign Up page and ability to create a Creator account and Benefit accounts    

#### Potential issues:
1) It looks like a "Name" attribute got added onto Benefit. Should users be able to specify a Benefit's name in addition to it's description when signing up?    
2) The test for throwing an error if Benefit description is longer than 420 chars is no longer working in the tests and I'm not sure why... I'll log this as an issue

### CODE REVIEW

File: app/pages/sign-up.tsx [hunk 1]
```diff
@@ -17,38 +54,94 @@ const SignUp = () => {
...
     const email = emailRef?.current?.value;
     const description = descriptionRef?.current?.value;
 
-    const benefits = benefitRefs.map((benefitRef) => {
+    let benefits = benefitRefs.map((benefitRef) => {
       return benefitRef?.current?.value;
     });
 
+    // Filter out blank benefits
+    benefits = benefits.filter((benefit) => benefit);
```
Comment (MikaelCarpenter) [inline app/pages/sign-up.tsx:68]: you can chain the `.map` and `.filter` (i.e. `.map(---).filter(---)`)so you don't have to do the `let` and reassigning variable thing

File: app/pages/sign-up.tsx [hunk 2]
```diff
@@ -17,38 +54,94 @@ const SignUp = () => {
...
+      program!.programId
+    );
+
+    // Create Creator account
+    const txn = new anchor.web3.Transaction();
+    txn.add(
+      program!.instruction.createAccount(
+        username!,
+        email!,
+        description!,
```
Comment (MikaelCarpenter) [inline app/pages/sign-up.tsx:88]: there may be a better way to handling the typing here. If these are nullable values, maybe we should handle the case when they are null by showing the user "This field is invalid" or something. 

Not important for this PR, but something we may want to get back to

File: programs/nft-club/src/lib.rs [hunk 3]
```diff
@@ -37,5 +37,11 @@ pub mod nft_club {
 
     pub fn update_subscription(ctx: Context<UpdateSubscription>) -> Result<()> {
         components::update_subscription(ctx)
+    pub fn delete_account(ctx: Context<DeleteAccount>) -> Result<()> {
```
Comment (MikaelCarpenter) [inline programs/nft-club/src/lib.rs:40]: we may be missing a character here, `update_subscription` should have a `}` that may have been deleted during the rebase

**Review by MikaelCarpenter — APPROVED** [review]
Looks good 👍

File: app/pages/sign-up.tsx (hunk 2 above)
Comment (KitanGarcia) [reply app/pages/sign-up.tsx:88]: You are completely right. Just wanted to get the progress in to unblock people. I just made an issue

**Pull request merged**